File: entities/license.go
package entities

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type VerificationStatus string

const (
	VerificationStatusUnverified VerificationStatus = "UNVERIFIED"
	VerificationStatusPending    VerificationStatus = "PENDING"
	VerificationStatusVerified   VerificationStatus = "VERIFIED"
	VerificationStatusRejected   VerificationStatus = "REJECTED"
)

type DoctorLicense struct {
	LicenseID           uuid.UUID          `gorm:"primaryKey" json:"license_id"`
	DoctorID            uuid.UUID          `gorm:"not null;index" json:"doctor_id"`
	RegistrationNumber  string             `gorm:"size:100;not null;index" json:"registration_number"`
	RegistrationCouncil string             `gorm:"size:255;not null" json:"registration_council"`
	RegistrationYear    int                `gorm:"not null" json:"registration_year"`
	IssueDate           *time.Time         `json:"issue_date,omitempty"`
	ExpiryDate          *time.Time         `json:"expiry_date,omitempty"`
	LicenseStatus       string             `gorm:"size:50;default:'ACTIVE'" json:"license_status"`
	VerificationStatus  VerificationStatus `gorm:"size:50;default:'UNVERIFIED';not null" json:"verification_status"`
	CreatedAt           time.Time          `gorm:"autoCreateTime" json:"created_at"`
	UpdatedAt           time.Time          `gorm:"autoUpdateTime" json:"updated_at"`
	DeletedAt           gorm.DeletedAt     `gorm:"index" json:"-"`
}

func (l *DoctorLicense) BeforeCreate(tx *gorm.DB) (err error) {
	if l.LicenseID == uuid.Nil {
		l.LicenseID = uuid.New()
	}
	return nil
}

File: python_backend/entities/license.py
import enum
import uuid
from sqlalchemy import Column, String, Integer, DateTime, Enum, ForeignKey
from sqlalchemy.sql import func
from database import Base

class VerificationStatus(str, enum.Enum):
    UNVERIFIED = "UNVERIFIED"
    PENDING = "PENDING"
    VERIFIED = "VERIFIED"
    REJECTED = "REJECTED"

class DoctorLicense(Base):
    __tablename__ = "doctor_licenses"

    license_id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    doctor_id = Column(String(36), ForeignKey("doctors.id", ondelete="CASCADE"), nullable=False, index=True)
    registration_number = Column(String(100), nullable=False, index=True)
    registration_council = Column(String(255), nullable=False)
    registration_year = Column(Integer, nullable=False)
    issue_date = Column(DateTime(timezone=True), nullable=True)
    expiry_date = Column(DateTime(timezone=True), nullable=True)
    license_status = Column(String(50), default='ACTIVE')
    verification_status = Column(Enum(VerificationStatus, name="verification_status_enum", native_enum=False), default=VerificationStatus.UNVERIFIED, nullable=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())
    deleted_at = Column(DateTime(timezone=True), index=True, nullable=True)
